### ISSUE TITLE
IPv4 Derived Bits Instance

### ISSUE THREAD

**Issue by LemonIceScream** [andrewthad/haskell-ip#29]
I have been working with the haskell-ip package for us with trees and I believe that the ordering on the derived instance of Bits is incorrect. The derived instance uses host ordering, however I found I am using and think of IPs in network order:

```
λ: testBit (Net.IPv4.ipv4 255 255 0 0) 0
False
λ: testBit (Net.IPv4.ipv4 255 255 0 0) 31
True
```

**Comment by andrewthad** [comment]
This is a good point. I will take a PR changing this behavior and adding haddock for the new instance explaining that it differs from `Word32`'s instance.

**Comment by LemonIceScream** [comment]
This can be closed. It was fixed in PR #31
